### ISSUE TITLE
How to set loss weight in chainer?

### ISSUE THREAD

**Issue by sharpstill** [chainer/chainer#3526]
I have already submit this issue to https://stackoverflow.com/questions/46647044/how-to-set-loss-weight-in-chainer but still can't wait for reply( I am hurry for my research paper submit)

First of all I narrate you about my question and situation. I want to do multi-label classification in chainer and my class imbalance problem is very serious.

In this cases I must slice the vector inorder to calculate loss function, For example, In multi-label classification, ground truth label vector most elements is 0, only few of them is 1, In this situation, directly use F.sigmoid_cross_entropy to apply all the 0/1 elements may cause training not convergence, So I decide to use a[[xx,xxx,...,xxx]] slice( a is chainer.Variable output by last FC layer) to slice specific elements to calculate loss function. In this case, because of label imbalance may cause rare class low classification performance, so I want to set rare gt-label variable high loss weight during back propagation, but set major label(occur too many in gt) variable low weight during back propagation.

How should I do it? What is your suggestion about multi-label imbalance class problem training in chainer?

**Comment by soskek** [comment]
How about using `F.sigmoid_cross_entropy` with `reduce='no'`?
It returns elementwise losses shaped `(batchsize, labels)` rather than scalar values as sum of losses. Then, you can weight each loss by multiplying weight arrays you prepare.
